### ISSUE TITLE
Country and state rules for shipping to U.S. possessions e.g. Puerto Rico

### ISSUE THREAD

**Issue by calendarboy** [woocommerce/woocommerce#22687]
The rules for addressing packages in the United States are determined by the United States Postal Service. FedEx and UPS follow the USPS rules. They all do. 

But, WooCommerce does not. 

I struggled with WooCommerce a couple of years  (2012 through 2014) to get a situation corrected. They finally did, but now they have changed it back. 

The pull-down list of States (with Country set for United States) does **not** include Puerto Rico, Virgin Islands, and the other US Possessions.

If you go to the USPS web site and follow the path to create a shipping label, you will see that USPS has under the State pull-down:

AS for American Samoa after Alaska

AA  (Armed Forces of Americas except Canada)
AE  (Armed Forces Europe)
AP  (Armed Forces Pacific)
And they have them between Arkansas and California. 

After DC, they have FM for Federated States of Micronesia
After Georgia they have Guam
MH for Marshall Islands
MP  (Northern Mariana Islands)
PR  for Puerto Rico
PW  for Palau
VI  for Virgin Islands

WooCommerce has American Samoa, Guam, Marshall Islands, Northern Mariana Islands, Puerto Rico, Virgin Islands (as United States (US) Virgin Islands) all listed as a different country in the Country pull-down.

All of these should all be in the State pull-down (when United States is selected for country). 

All three of the ArmedForces are at the end of the States. They should be in **alphabetical order**. 

WooCommerce doesn't have Federated States of Micronesia or Palau listed at all in Country or State. 

I was stunned that it took years to get John to change these. Even more shocked that someone changed them all back. 

It does not matter that International Standard list of countries might have them listed as separate "countries". We are talking about COMMERCE. 

What matters is how the shipping companies treat shipments. All of the above should be changed back to how they are treated by USPS.

![screen shot 2019-02-07 at 4 07 06 pm](https://user-images.githubusercontent.com/5973789/52451123-9282ab00-2af2-11e9-8625-8aa0fb045d4e.png)

**Comment by mikejolley** [comment]
I don't have the correct answer for this but you're basing your argument on how you, the shipper, selects states when you're shipping out of the US.

If you lived in Puerto Rico, which is not a US state, would you select "United States" from the country dropdown, or look for your actual country, "Puerto Rico".

Checkout should be aimed at the customer, not you, the shipper. And if a service such as USPS expects something different, the plugin handling that carrier should do the translation.

**Comment by calendarboy** [comment]
Mike,

I did not list this issue on GitHub for my “convenience”. I did it because the way WooCommerce handles it is actually wrong!

You and I went through this in 2013, 2014, and more. You kept telling me how Puerto Rico, Samoa, Virgin Islands are listed separately from the United States on some International Standards list. Back then WooCommerce even had the Military addresses under “Country” instead of “State”.

I actually got you to change this so that the Possessions were in the State pull-down. But, it got moved back.

I even eventually got you to change the “State” names to the two letter Postal abbreviations where WooC had been spelling everything out like “Washington, District of Columbia”.

You said, “Basing the argument on how you, the shipper…..”

No. No No.  

If I were in Puerto Rico, or any U.S. State or U.S. Possession, buying something online to be delivered to an address in Puerto Rico, regardless whether it was with FedEx, UPS, or the US Postal Service, I would select “United States” as the Country, then “Puerto Rico”, etc. as the State.  

Just like the screen shot from US Postal Serveice that I included which showed United States for Country and Virgin Islands for State. 

No, Puerto Rico, the US Virgin Islands, Washington, DC, American Samoa, and others do NOT have Statehood. 
But, they are U.S. Possessions. Shipments to any of them are domestic, the same as shipping to California, Florida, or any State.  They even have U.S. Zip Codes. They have the 2-letter “State” abbreviations. The residents have U.S. Citizenship. They pay U.S. taxes. 

And, NO, someone in Puerto Rico, the US Virgin Islands, Washington, DC, American Samoa, etc. would not look in the pull-down for Country. Their Country is the United States. Someone unfamiliar with the United States might look in Country, but not people who live in those possessions. 

It is not a matter of it being “easier” or more convenient for me. Anyone with any experience having items shipped to them in U.S. Possessions, expects to put United States as the Country and “Virgin Islands”, “Puerto Rico”, etc. as the State. They

There are actually quite a number of people who have no idea that Puerto Rico, Virgin Islands, etc. are U.S. Possessions. The hurricane damage to PR brought to light that many Americans had no idea Puerto Ricans were U.S. Citizens. 
Therefore, for the clueless, the USPS includes the U.S. Virgin Islands, Guam, American Samoa, and so on in BOTH the International calculator and the U.S. calculator. But, the rates in both places are the same for U.S. possessions.

Customers ordering from the U.S. Possessions expect to find their territory listed as a State. It is for clarity for the customer. Do not dismiss it as something for my convenience. 

Your last paragraph… Very, very annoying. Second time I’ve heard that over the past two months. WooCommerce prefers to do it whatever way they feel like and you can fix it further down the path. 
> if a service such as USPS expects something different, the plugin handling that carrier should do the translation
> 

Really? Get some plugin to fix things WooCommerce does because . . .  why?

LABELS: As long as I have had WooCommerce, I have exported orders to a spreadsheet, inserted commands in columns between items for the shipping addresses, copied all of the spreadsheet to a text document, then imported all of it into Adobe InDesign. 

Basically, I hand-make the labels. We pack the orders and put postage on with a postage meter. 

For some years, I had to correct data for labels changing the written-out State names and District of Columbia to the 2-letter abbreviations. Manually. 

Now I have to take “United States” out of every label because you have hard-coded Country into every address.


Sincerely,
Jim Maynard
1132 Augusta
Ashland, Oregon 97520

Cell 541-951-7768

**Comment by mikejolley** [comment]
I'm not from the US nor Puerto Rico so as I said, I don't know the answer. This issue has only come up twice, this being the 2nd time I can find on GitHub, for "Puerto Rico" anyway, so does not seem to be a common concern or major issue.. 

The main problem is this:

> And if a service such as USPS expects something different, the plugin handling that carrier should do the translation.

Most services, at least those I've come across, expect ISO country codes. Turning this into a state, at the technical level, could break this. Thats why I'm saying, with the current code, USPS etc extensions can change the ISO country code into whatever that service needs. They already do this. Changing it **has implications for payment gateways and shipping services.**

I'd be curious to see how sites like Amazon present this.

When I tried the address form on Amazon, "Puerto Rico" was listed as a country. This may be because I'm not in the US so someone else will need to check. eCommerce sites like Amazon (customer facing) would be a better reference than USPS _which is store owner/shipper facing._ Several examples would be helpful to backup your statements.

**Comment by calendarboy** [comment]
Mike,

Just for giggles, check the dates on the emails in the two attached screenshots (upper, right-hand corner). 


So, I get it. WooCommerce simply isn’t flexible enough to list States and Possessions based on how American commerce works. At least you still have the Armed Forces in there. When I started using WooC, they were listed under Countries. 

Jim

**Comment by GlennMartin1** [comment]
@mikejolley

I live in the US, and I can confirm that both **Amazon** and **eBay** list **Puerto Rico, American Samoa**, and **Virgin Islands** as a **country**.

**Comment by mikejolley** [comment]
Thanks @GlennMartin1, someone on our team also confirmed Amazon lists them as countries not states.

Shopify lists them as states rather than countries.

ISO aparently lists both, e.g. PR:

> This code is part of collection(s) Country Codes Collection
> No subdivisions relevant for this standard. Included also as a subdivision of the United States (US-PR).

@bekarice @rrennick do you perhaps have any thoughts on this one?

FYI If anyone wants to customise this, states can be changed via the filters: https://docs.woocommerce.com/document/addmodify-states/

**Comment by rrennick** [comment]
I'm not in the US either so I looked up how I would ship to PR & found that Canada Post ships to Puerto Rico with its own country code: https://www.canadapost.ca/tools/pg/manual/PGIntDestDetails-e.asp?table=tblCountry&txtLetter=PR

**Comment by bekarice** [comment]
@mikejolley afaik handling for territories depends on the shipping service -- USPS uses these as states (given it's handling them as domestic packages), while other shippers treat them as countries. Shipstation has a [good overview here](https://help.shipstation.com/hc/en-us/articles/206640527-How-do-I-ship-to-US-territories-like-Puerto-Rico-and-Guam-), and [Ordero seems to confirm it](https://support.ordoro.com/what-is-the-address-format-to-ship-to-puerto-rico/). For example, UPS [does not include territories as US states](https://www.ups.com/worldshiphelp/WS19/ENU/AppHelp/Codes/State_Province_Codes.htm).

As a result, given USPS is the only one viewing them as _states_ (since they're considered "domestic" to USPS in that case), I think any USPS integrations should filter the states / countries lists themselves for a format USPS expects. (This could probably be done globally when a USPS plugin is enabled, even if it's not the selected shipping method at checkout, as for most carriers, the package will be picked up by USPS after customs; I think the only exceptions are USP, FedEx, DHL.)

Given all other shipping services expect to see these territories listed outside the US as separate countries, that should likely be the default within WC core.

**Comment by mikejolley** [comment]
Thanks @bekarice and @rrennick. Those links are very helpful.

Flagging this with @woocommerce/hydra and @woocommerce/extendables just in case they want to look into that for our USPS integrations.

Since this is customisable I agree it's best to leave this alone for now unless there is urgent need to modify the country/state system for these places.

I'll modify the issue title so it's easier to refer back to in future.

Thanks all!

**Comment by taxilian** [comment]
The most frustrating thing about this decision is that stamps.com and doubtless some other shipping systems give you an error if you try to erroneously list US territories as their own country. It baffles me that anyone thinks this is a good idea -- I have talked to people in PR and Guam and they have no idea why so many list it that way either.

They aren't separate countries. They dont' think of themselves as separate countries. Many of them are exasperated that so many elsewhere in the US don't realize that they are part of the US.

District of Columbia is also a US territory, not a state  -- yet it is listed under states. it has exactly the same status, that of a "territory".

**Comment by chadwikchugh** [comment]
This still isn't resolved.  I can not provide rates to customers in PR because Shipstation (and USPS, Stamps.com, UPS, FedEx, etc) do not list PR as a country, but as a State.  Woocommerce does not allow for this.  At one point they did and we had no issues (so it seemed.)

**Comment by taxilian** [comment]
This is the crap "workaround" that I'm using for now:

```
add_filter( "woocommerce_states", "usa_woocommerce_states" );

function usa_woocommerce_states( $states ) {

	$states["US"]["GU"] = "Guam";
	$states["US"]["PR"] = "Puerto Rico";
	$states["US"]["AS"] = "American Samoa";
	$states["US"]["MP"] = "Northern Mariana Islands";
	$states["US"]["PW"] = "Palau";
	$states["US"]["VI"] = "U.S. Virgin Islands";
	$states["US"]["MH"] = "Marshall Islands";

	asort($states);
	return $states;
}
```

Note that doesn't really "fix" the problem, but it at least makes it possible to get a workaround; using that if I select e.g. "PR" as a state then I get rates from USPS and if I select it as a country I get rates from UPS.

**Comment by chadwikchugh** [comment]
taxilian, I did something similar, just added PR as a state to allow for USPS.  

This really needs resolved for all the US territories.  What's odd is that years ago we had shipments going through to PR without a hitch, only recently have we had difficulties with this.
